src/Factories/FunFunFactory.php

```diff
@@ -1372,4 +1373,11 @@ public function setContentProvider( ContentProvider $contentProvider ): void {
 	private function getContentProvider(): ContentProvider {
 		return $this->pimple['content_provider'];
 	}
+
+	public function newMailTemplateFilenameTraversable(): \Traversable {
```

Comment:
Can you elaborate on why you want it to be more specific? My reasoning: being more specific here would mean using `IteratorAggregate`. Then if you change the implementation, suddenly you have a breaking change here. And why would the consumer care it's an `IteratgorAggregate`? Suppose this was a `Generator`, then I think it would be sensible to type hint as `Iterator`, since the consumer benefits, and most refactorings would result in an `Iterator`.

For use statements I'm using the PHPStorm default: inline when it's a ref to a top level NS class, with use statement if it is not. This default makes sense to me and I'm not sure why you'd want to deviate.